app/services/course_search_service.rb

```diff
@@ -51,44 +52,64 @@ def call
 
 private
 
-  def sites_with_distance_from_origin
+  def distance_with_university_area_adjustment
+    university_provider_type = Provider.provider_types[:university]
+    university_location_area_radius = 10
+    <<~EOSQL.gsub(/^[\s\t]*/, "").gsub(/[\s\t]*\n/, " ").strip
+      (CASE
+        WHEN provider.provider_type = '#{university_provider_type}'
+          THEN (distance - #{university_location_area_radius})
+        ELSE distance
+      END) as boosted_distance
+    EOSQL
+  end
+
+  def locatable_sites
     site_status = SiteStatus.arel_table
     sites = Site.arel_table
 
-    # Create virtual table with sites and site statuses
-    sites_with_status = site_status.join(sites).on(site_status[:site_id].eq(sites[:id]))
-
-    # only want new and running sites
-    new_and_running_sites = sites_with_status.where(site_status[:status].in(%w[new_status running]))
+    # Only running and published site statuses
+    running_and_published_criteria = site_status[:status].eq(SiteStatus.statuses[:running]).and(site_status[:publish].eq(SiteStatus.publishes[:published]))
```

Comment:
I really like how you're breaking down the query into chunks. Would it not be too much if we went just a tiny but further with this one?

```ruby
     running_criteria = site_status[:status].eq(SiteStatus.statuses[:running])
     published_criteria = site_status[:publish].eq(SiteStatus.publishes[:published])
```

And of course you'd need one more `.where(published_criteria)` below.

Reply:
I'm not sure.

What I really want to enshrined is

https://github.com/DFE-Digital/teacher-training-api/blob/a97b30d65a5e2f357e696d0a5f406e6d1b144fb2/app/models/site_status.rb#L59

Which is a meaningful chunk.. not sure about calling it `findable_criteria` there is already too much unfindable things around.

The given logic got to go through this 
- [findable](https://github.com/DFE-Digital/teacher-training-api/blob/b8c807c3e2f7eb5d2b3438f3ea65be54f9d8a2fb/app/services/course_search_service.rb#L15)

which is this
- [findable](https://github.com/DFE-Digital/teacher-training-api/blob/fdb35d6428d7ad39ad570fe39cba2086ebde8318/app/models/course.rb#L182)

which is really back to this
- [findable](https://github.com/DFE-Digital/teacher-training-api/blob/a97b30d65a5e2f357e696d0a5f406e6d1b144fb2/app/models/site_status.rb#L59)

then its
- [delimited](https://github.com/DFE-Digital/teacher-training-api/blob/b8c807c3e2f7eb5d2b3438f3ea65be54f9d8a2fb/app/services/course_search_service.rb#L29)

before it hits `findable_criteria` and then filtered